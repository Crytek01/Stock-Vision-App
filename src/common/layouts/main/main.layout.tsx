import { Footer } from "@/common/components/footer/footer.component";
import { NavBar } from "@/common/components/navbar/navbar.component";
import { FunctionComponent } from "react";

interface IMainLayoutProps {
  children: React.ReactNode;
}

export const MainLayout: FunctionComponent<IMainLayoutProps> = ({
  children,
}) => {
  return (
    <>
      <header>
        <NavBar />
      </header>

      <main>{children}</main>

      <Footer />
    </>
  );
};
