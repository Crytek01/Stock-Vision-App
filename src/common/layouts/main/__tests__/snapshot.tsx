import { render } from "@testing-library/react";
import { MainLayout } from "../main.layout";

it("renders MainLayout component", () => {
  const { container } = render(
    <MainLayout>
      <div>Hi</div>
    </MainLayout>
  );
  expect(container).toMatchSnapshot();
});
