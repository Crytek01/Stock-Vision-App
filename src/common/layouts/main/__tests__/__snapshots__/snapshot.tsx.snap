// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainLayout component 1`] = `
<div>
  <header>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Stock Vision
        </h6>
      </div>
    </header>
  </header>
  <main>
    <div>
      Hi
    </div>
  </main>
  <footer
    class="footer"
  >
    <p>
      © 
      2023
       Stock Vision. All Rights Reserved.
    </p>
    <p
      class="footerLastText"
    >
      Made with ❤️ by Erick Badilla
    </p>
  </footer>
</div>
`;
